kernel/src/debug.rs

```diff
@@ -342,22 +349,28 @@ impl DebugWriter {
     }
 }
 
-impl hil::uart::Client for DebugWriter {
-    fn transmit_complete(&self, buffer: &'static mut [u8], _error: hil::uart::Error) {
+impl hil::uart::TransmitClient for DebugWriter {
+    fn transmitted_buffer(&self, buffer: &'static mut [u8], tx_len: usize, _rcode: ReturnCode) {
         // Replace this buffer since we are done with it.
         self.output_buffer.replace(buffer);
 
-        let written_length = self.active_len.get();
-        self.active_len.set(0);
+        // Mark how many bytes outstanding so we don't overwrite buffer
+        // in transmit calls.
+        let goal_length = self.active_len.get();
+        let remainder = goal_length - tx_len;
```

Comment:
Are we sure that `active_len => tx_len`?

Reply:
Yes; active_len is the length that was passed to the call to transmit_buffer, tx_len is the number of bytes actually transmitted. The interface definition is such that tx_len <= active_len.

Reply:
However, if there is a bug in the callee, then this might not be true -- if tx_len > active_len, that's definitely an error, which should be caught carefully rather than go kaboom.